modules/masterbar/masterbar.php

```diff
@@ -0,0 +1,823 @@
+				'label' => __( 'Blog Posts', 'jetpack' ),
+			),
+			array(
+				'url'   => 'https://wordpress.com/post/' . esc_attr( $this->primary_site_slug ),
+				'id'    => 'wp-admin-bar-new-post',
+				'label' => _x( 'Add', 'admin bar menu new item label', 'jetpack' ),
+			)
+		);
+
+		$wp_admin_bar->add_menu( array(
+			'parent' => 'publish',
+			'id'     => 'new-post',
+			'title'  => $posts_title,
+			'meta'   => array(
+				'class' => 'inline-action',
+			),
+		) );
+
+		// Pages
+		$pages_title = $this->create_menu_item_pair(
+			array(
+				'url'   => 'https://wordpress.com/pages/' . esc_attr( $this->primary_site_slug ),
+				'id'    => 'wp-admin-bar-edit-page',
+				'label' => __( 'Pages', 'jetpack' ),
+			),
+			array(
+				'url'   => 'https://wordpress.com/page/' . esc_attr( $this->primary_site_slug ),
+				'id'    => 'wp-admin-bar-new-page',
+				'label' => _x( 'Add', 'admin bar menu new item label', 'jetpack' ),
+			)
+		);
+
+		$wp_admin_bar->add_menu( array(
+			'parent' => 'publish',
+			'id'     => 'new-page',
+			'title'  => $pages_title,
+			'meta'   => array(
+				'class' => 'inline-action',
+			),
+		) );
+
+		// Portfolio
+		$portfolios_title = $this->create_menu_item_pair(
+			array(
+				'url'   => 'https://wordpress.com/types/jetpack-portfolio/' . esc_attr( $this->primary_site_slug ),
+				'id'    => 'wp-admin-bar-edit-portfolio',
+				'label' => __( 'Portfolio', 'jetpack' ),
+			),
+			array(
+				'url'   => 'https://wordpress.com/edit/jetpack-portfolio/' . esc_attr( $this->primary_site_slug ),
+				'id'    => 'wp-admin-bar-new-portfolio',
+				'label' => _x( 'Add', 'admin bar menu new item label', 'jetpack' ),
+			)
+		);
+
+		$wp_admin_bar->add_menu( array(
+			'parent' => 'publish',
+			'id'     => 'new-jetpack-portfolio',
+			'title'  => $portfolios_title,
+			'meta'   => array(
+				'class' => 'inline-action',
+			),
+		) );
+
+		if ( current_user_can( 'edit_theme_options' ) ) {
+			// Look and Feel group
+			$wp_admin_bar->add_group( array(
+				'parent' => 'blog',
+				'id'     => 'look-and-feel',
+			) );
+
+			// Look and Feel header
+			$wp_admin_bar->add_menu( array(
+				'parent' => 'look-and-feel',
+				'id'     => 'look-and-feel-header',
+				'title'  => _x( 'Personalize', 'admin bar menu group label', 'jetpack' ),
+				'meta'   => array(
+					'class' => 'ab-submenu-header',
+				),
+			) );
+
+			if ( is_admin() ) {
+				// In wp-admin the `return` query arg will return to that page after closing the Customizer
+				$customizer_url = add_query_arg( array( 'return' => urlencode( site_url( $_SERVER['REQUEST_URI'] ) ) ), wp_customize_url() );
+			} else {
+				// On the frontend the `url` query arg will load that page in the Customizer and also return to it after closing
+				// non-home URLs won't work unless we undo domain mapping since the Customizer preview is unmapped to always have HTTPS
+				global $current_blog;
+				$current_page = 'https://' . $current_blog->domain . $_SERVER['REQUEST_URI'];
```

Comment:
When viewing a single post/page, looks like $current_blog is not set as you expected.  I am getting 
`Notice: Trying to get property of non-object in /home/dereksma/public_html/wp-content/plugins/jetpack/modules/masterbar/masterbar.php on line 666` on single post views on the front end.